inc/functions-home.php

```diff
@@ -164,6 +143,17 @@ function update_page_delete_transient() {
 	}
 }
 
+/**
+ * @param $date
+ * @param string $format
+ *
+ * @return bool
+ */
+function validate_date( $date, $format = 'Y-m-d H:i:s' ) {
```

Comment:
Hi @domingobishop - I'd suggest changing this function slightly. I'll send you an example implementation that uses a regex shortly...

Reply:
Hey @domingobishop - what do you think of this as an alternative approach. To be honest, I'd be quite happy with the original approach too.

```php 
<?php

$example_dates = [
    '2017-10-15T12:39', // Correct
    '00o0-10-15T12:39', // Invalid year
    '2017-45-15T25:39', // Invalid hours
    '2017-10-15T12:61', // Invalid seconds
    '2017-10-32T12:39', // Invalid day
    '2017-10-32', // Missing time
];

function validate_date($date, $format = 'Y-m-d\TH:i')
{
    if (preg_match('/^\d{4}-\d{2}\-\d{2}T\d{2}:\d{2}/', $date)) { // Is in correct format
        return ((bool)strtotime($date)); // Is a valid date
    }
    return false;
}

foreach ($example_dates as $date) {
    echo $date . ' is ' . (validate_date($date) ? 'Good' : 'Bad') . '<br>';
}
```